src/main/java/org/aksw/iguana/cc/suite/IguanaSuiteParser.java

```diff
@@ -0,0 +1,260 @@
+                JsonNode node = jp.getCodec().readTree(jp);
+                if (node.isTextual()) {
+                    final var connectionName = node.asText();
+                    if (!connections.containsKey(connectionName))
+                        throw new IllegalStateException(MessageFormat.format("Unknown connection name: {0}", connectionName));
+                    return connections.get(connectionName);
+                } else {
+                    ConnectionConfig connectionConfig = ctxt.readValue(jp, ConnectionConfig.class);
+                    if (connections.containsKey(connectionConfig.name()))
+                        assert connections.get(connectionConfig.name()) == connectionConfig;
+                    else connections.put(connectionConfig.name(), connectionConfig);
+                    return connectionConfig;
+                }
+            }
+        }
+
+        class HumanReadableDurationDeserializer extends StdDeserializer<Duration> {
+
+            public HumanReadableDurationDeserializer() {
+                this(null);
+            }
+
+            protected HumanReadableDurationDeserializer(Class<?> vc) {
+                super(vc);
+            }
+
+            @Override
+            public Duration deserialize(JsonParser jp, DeserializationContext ctxt) throws IOException {
+                var durationString = jp.getValueAsString()
+                        .toLowerCase()
+                        .replaceAll("\\s+", "")
+                        .replace("years", "y")
+                        .replace("year", "y")
+                        .replace("months", "m")
+                        .replace("month", "m")
+                        .replace("weeks", "w")
+                        .replace("week", "w")
+                        .replace("days", "d")
+                        .replace("day", "d")
+                        .replace("mins", "m")
+                        .replace("min", "m")
+                        .replace("hrs", "h")
+                        .replace("hr", "h")
+                        .replace("secs", "s")
+                        .replace("sec", "s")
+                        .replaceFirst("(\\d+d)", "P$1T");
+                if ((durationString.charAt(0) != 'P')) durationString = "PT" + durationString;
+                return Duration.parse(durationString);
+            }
+        }
+
+        mapper = new ObjectMapper(factory).registerModule(new JavaTimeModule())
+                .registerModule(new SimpleModule()
+                        .addDeserializer(DatasetConfig.class, new DatasetDeserializer())
+                        .addDeserializer(ConnectionConfig.class, new ConnectionDeserializer())
+                        .addDeserializer(Duration.class, new HumanReadableDurationDeserializer()));
+
+        final String suiteID = Instant.now().getEpochSecond() + "-" + Integer.toUnsignedString(input.hashCode()); // convert to unsigned, so that there is no double -- minus in the string
```

Comment:
remind me: Instant.now() was wall and non-monotonic time, right? would make to replace them all.

Reply:
Yes I think so. But I don't think `Instant.now()` should be an issue here, because it's not used here to measure elapsed time, it's only used as a timestamp. On the other hand `System.nanoTime()` counts the time from an arbitrary point of time, so if it's used as a "timestamp", its value would be quite meaningless other than being some random numbers: https://docs.oracle.com/en/java/javase/17/docs/api/java.base/java/lang/System.html#nanoTime()

Reply:
But I can change it if you want